proposals/lip-Introduce_PoA_module.md

```diff
@@ -0,0 +1,614 @@
+                    "type": "array",
+                    "fieldNumber": 1,    
+                    "items": {
+                        "type": "object",
+                        "required": ["address", "weight"],
+                        "properties": {
+                            "address": {
+                                "dataType": "bytes",
+                                "fieldNumber": 1
+                            },
+                            "weight": {
+                                "dataType": "uint64",
+                                "fieldNumber": 2
+                            }
+                        }
+                    }
+                },
+                "threshold": {
+                    "dataType": "uint64",
+                    "fieldNumber": 2
+                },
+                "validatorsUpdateNonce": {
+                    "dataType": "uint32",
+                    "fieldNumber": 3
+                }
+            },
+            "required": ["newValidators", "threshold", "validatorsUpdateNonce"]
+        }
+        ```
+        *   The `weights` argument is set to `[ validator.weight for validator in snapshotStore(0).validators]`.
+        *   The `threshold` argument is set to `snapshotStore(0).threshold`.
+
+##### Execution 
+
+Let `trs` be a transaction with module ID `MODULE_ID_POA` and command ID `COMMAND_ID_UPDATE_AUTHORITY` to be processed. 
+Then processing `trs` has the following effect:
+
+*   The array `snapshotStore(2).validators` is set to `trs.params.newValidators`.
+*   The property `snapshotStore(2).threshold` is set to `trs.params.threshold`.
+*   The property `chainProperties.validatorsUpdateNonce` is set to `trs.params.validatorsUpdateNonce + 1`.
+
+### Internal Function
+
+#### shuffleValidatorsList
+
+A function to reorder the list of validators as specified in [LIP 0003][lip3].
+
+##### Parameters
+The function has the following input parameters in the order given below:
+
+* `validatorsAddresses`: An array of pairwise distinct 20-byte addresses.
+* `randomSeed`: A 32-byte value representing a random seed.
+
+##### Returns
+This function returns an array of bytes with the re-ordered list of addresses.
+
+##### Execution
+
+```python
+shuffleValidatorsList(validatorsAddresses, randomSeed)
+    
+    validatorsList = []
+    for every address in validatorsAddresses:
+        validator.address = address
+        validator.roundHash = hash(randomSeed || address)
+        push validator to validatorsList
+    
+    # Reorder the validator list in lexicographical order first by roundHash, then by address
+    sort validatorsList in increasing order where validator1 < validator2 if 
+    (validator1.roundHash < validator2.roundHash) || 
+    ((validator1.roundHash == validator2.roundHash) && (validator1.address < validator2.address))
+    
+    return [address for every address in validatorsList]
+```
+where the `hash` function is SHA256 and `||` is byte concatenation.
```

Comment:
```suggestion
where `hash` is SHA256 and `||` is byte concatenation.
```

Reply:
`hash` is the name of the function, we do not talk about the concept of a hash function here

Reply:
changed